pkg/apis/options/legacy_options.go

```diff
@@ -509,6 +509,10 @@ type LegacyProvider struct {
 	UserIDClaim                        string   `flag:"user-id-claim" cfg:"user_id_claim"`
 	AllowedGroups                      []string `flag:"allowed-group" cfg:"allowed_groups"`
 
+	SkipAudCheckWhenMissing       bool     `flag:"skip-aud-check-when-missing" cfg:"skip_aud_check_when_missing"`
+	AudienceVerificationClaim     string   `flag:"audience-verification-claim" cfg:"audience_verification_claim"`
```

Comment:
I wonder if this is simpler if just setup as `oidc-aud-claim` that matches `oidc-email-claim` and `oidc-groups-claim` we have to support those claims that some providers have non-standard (similar to your issue with AWS Cognito `aud` vs `client_id` (?) I think you said.